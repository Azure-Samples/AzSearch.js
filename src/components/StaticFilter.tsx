import * as React from "react";
import { PropsType } from "../containers/StaticFilterContainer";
import * as objAssign from "object-assign";
import { Store } from "@microsoft/azsearchstore";
import { defaultCss } from "../utils/css";

export type State = {};

class StaticFilter extends React.PureComponent<PropsType, State> {
  render() {
    const { filters, beforeFirstRequest, onFilterChange, activeFilter, title, filterKey } = this.props;
    let css = objAssign({}, defaultCss, this.props.css);

    if (beforeFirstRequest) return <div></div>;

    let options = filters.map((filter, index) => {
      return <option key={index} selected={filter.filter === activeFilter} value={filter.filter}>{filter.displayName ? filter.displayName : filter.filter}</option>;
    });

    let onChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
      onFilterChange(event.target.value);
    };

    let container = <div className={css.sorting__sortBy}>
      <div className={css.searchFacets__facetHeaderContainer}>
          <h4 className={css.searchFacets__facetHeader}>
              <a data-toggle="collapse" className={css.searchFacets__facetHeaderLink}  >
                  <span className={css.searchFacets__facetHeaderIconOpen} aria-hidden="true"></span> { title ? title : filterKey}
              </a>
          </h4>
      </div>
      <div className={css.searchFacets__facetControlContainer}>
        <div className={css.searchFacets__facetControlList}>
          <select className={css.sorting__sortByControl} onChange={onChange}>
            {options}
          </select>
        </div>
      </div>
    </div>;

    return (container);
  }
}

export default StaticFilter;