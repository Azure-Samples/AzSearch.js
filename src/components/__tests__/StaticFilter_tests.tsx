import * as React from "react";
import StaticFilter from "../StaticFilter";
import { shallow } from "enzyme";

const filters = [
    { displayName: "Any", filter: "" },
    { displayName: "House", filter: "type eq 'House'" },
    { displayName: "Apartment", filter: "type eq 'Apartment'" }
];
const defaultFilter = "";
const title = "Home Type";

describe("<StaticFilter/>", () => {
    it("renders against dummy snapshot", () => {

        expect(shallow(
            <StaticFilter  css={null} filters={filters} activeFilter={defaultFilter} filterKey={"type"} beforeFirstRequest={false} onFilterChange={function(foo){}} title={title} />
        )).toMatchSnapshot();
    });
});