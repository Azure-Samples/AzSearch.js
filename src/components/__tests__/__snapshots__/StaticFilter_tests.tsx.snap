// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StaticFilter/> renders against dummy snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="sorting__sortyBy panel-body"
>
    <div
        className="searchResults__facetHeader-container panel-heading"
    >
        <h4
            className="searchResults__facetHeader panel-title"
        >
            <a
                className="searchResults__facetHeader-link"
                data-toggle="collapse"
            >
                <span
                    aria-hidden="true"
                    className="searchResults__facetHeader-icon--open indicator glyphicon glyphicon-triangle-bottom"
                />
                 
                Home Type
            </a>
        </h4>
    </div>
    <div
        className="searchResults__facetControl-container panel-collapse collapse in"
    >
        <div
            className="searchResults__facetControl-list list-group"
        >
            <select
                className="sorting__sortyByControl list-group-item"
                onChange={[Function]}
            >
                <option
                    selected={true}
                    value=""
                >
                    Any
                </option>
                <option
                    selected={false}
                    value="type eq 'House'"
                >
                    House
                </option>
                <option
                    selected={false}
                    value="type eq 'Apartment'"
                >
                    Apartment
                </option>
            </select>
        </div>
    </div>
</div>,
  "nodes": Array [
    <div
      className="sorting__sortyBy panel-body"
>
      <div
            className="searchResults__facetHeader-container panel-heading"
      >
            <h4
                  className="searchResults__facetHeader panel-title"
            >
                  <a
                        className="searchResults__facetHeader-link"
                        data-toggle="collapse"
                  >
                        <span
                              aria-hidden="true"
                              className="searchResults__facetHeader-icon--open indicator glyphicon glyphicon-triangle-bottom"
                        />
                         
                        Home Type
                  </a>
            </h4>
      </div>
      <div
            className="searchResults__facetControl-container panel-collapse collapse in"
      >
            <div
                  className="searchResults__facetControl-list list-group"
            >
                  <select
                        className="sorting__sortyByControl list-group-item"
                        onChange={[Function]}
                  >
                        <option
                              selected={true}
                              value=""
                        >
                              Any
                        </option>
                        <option
                              selected={false}
                              value="type eq 'House'"
                        >
                              House
                        </option>
                        <option
                              selected={false}
                              value="type eq 'Apartment'"
                        >
                              Apartment
                        </option>
                  </select>
            </div>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <StaticFilter
        activeFilter=""
        beforeFirstRequest={false}
        css={null}
        filterKey="type"
        filters={
                Array [
                        Object {
                          "displayName": "Any",
                          "filter": "",
                        },
                        Object {
                          "displayName": "House",
                          "filter": "type eq 'House'",
                        },
                        Object {
                          "displayName": "Apartment",
                          "filter": "type eq 'Apartment'",
                        },
                      ]
        }
        onFilterChange={[Function]}
        title="Home Type"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StaticFilter {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "activeFilter": "",
          "beforeFirstRequest": false,
          "css": null,
          "filterKey": "type",
          "filters": Array [
            Object {
              "displayName": "Any",
              "filter": "",
            },
            Object {
              "displayName": "House",
              "filter": "type eq 'House'",
            },
            Object {
              "displayName": "Apartment",
              "filter": "type eq 'Apartment'",
            },
          ],
          "onFilterChange": [Function],
          "title": "Home Type",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="sorting__sortyBy panel-body"
>
          <div
                    className="searchResults__facetHeader-container panel-heading"
          >
                    <h4
                              className="searchResults__facetHeader panel-title"
                    >
                              <a
                                        className="searchResults__facetHeader-link"
                                        data-toggle="collapse"
                              >
                                        <span
                                                  aria-hidden="true"
                                                  className="searchResults__facetHeader-icon--open indicator glyphicon glyphicon-triangle-bottom"
                                        />
                                         
                                        Home Type
                              </a>
                    </h4>
          </div>
          <div
                    className="searchResults__facetControl-container panel-collapse collapse in"
          >
                    <div
                              className="searchResults__facetControl-list list-group"
                    >
                              <select
                                        className="sorting__sortyByControl list-group-item"
                                        onChange={[Function]}
                              >
                                        <option
                                                  selected={true}
                                                  value=""
                                        >
                                                  Any
                                        </option>
                                        <option
                                                  selected={false}
                                                  value="type eq 'House'"
                                        >
                                                  House
                                        </option>
                                        <option
                                                  selected={false}
                                                  value="type eq 'Apartment'"
                                        >
                                                  Apartment
                                        </option>
                              </select>
                    </div>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="sorting__sortyBy panel-body"
>
          <div
                    className="searchResults__facetHeader-container panel-heading"
          >
                    <h4
                              className="searchResults__facetHeader panel-title"
                    >
                              <a
                                        className="searchResults__facetHeader-link"
                                        data-toggle="collapse"
                              >
                                        <span
                                                  aria-hidden="true"
                                                  className="searchResults__facetHeader-icon--open indicator glyphicon glyphicon-triangle-bottom"
                                        />
                                         
                                        Home Type
                              </a>
                    </h4>
          </div>
          <div
                    className="searchResults__facetControl-container panel-collapse collapse in"
          >
                    <div
                              className="searchResults__facetControl-list list-group"
                    >
                              <select
                                        className="sorting__sortyByControl list-group-item"
                                        onChange={[Function]}
                              >
                                        <option
                                                  selected={true}
                                                  value=""
                                        >
                                                  Any
                                        </option>
                                        <option
                                                  selected={false}
                                                  value="type eq 'House'"
                                        >
                                                  House
                                        </option>
                                        <option
                                                  selected={false}
                                                  value="type eq 'Apartment'"
                                        >
                                                  Apartment
                                        </option>
                              </select>
                    </div>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <StaticFilter
    activeFilter=""
    beforeFirstRequest={false}
    css={null}
    filterKey="type"
    filters={
        Array [
            Object {
              "displayName": "Any",
              "filter": "",
            },
            Object {
              "displayName": "House",
              "filter": "type eq 'House'",
            },
            Object {
              "displayName": "Apartment",
              "filter": "type eq 'Apartment'",
            },
          ]
    }
    onFilterChange={[Function]}
    title="Home Type"
/>,
}
`;
